tests/components/husqvarna_automower_ble/test_config_flow.py

```diff
@@ -0,0 +1,176 @@
+from homeassistant.config_entries import SOURCE_USER
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResultType
+
+from . import (
+    AUTOMOWER_MISSING_SERVICE_SERVICE_INFO,
+    AUTOMOWER_SERVICE_INFO,
+    AUTOMOWER_UNNAMED_SERVICE_INFO,
+    AUTOMOWER_UNSUPPORTED_GROUP_SERVICE_INFO,
+)
+
+from tests.components.bluetooth import inject_bluetooth_service_info
+
+pytestmark = pytest.mark.usefixtures("mock_setup_entry")
+
+
+async def test_user_selection(
+    hass: HomeAssistant,
+) -> None:
+    """Test we can select a device."""
+
+    inject_bluetooth_service_info(hass, AUTOMOWER_SERVICE_INFO)
+    inject_bluetooth_service_info(hass, AUTOMOWER_UNNAMED_SERVICE_INFO)
+    await hass.async_block_till_done(wait_background_tasks=True)
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}
+    )
+    assert result["type"] is FlowResultType.FORM
+    assert result["step_id"] == "user"
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        user_input={"address": "00000000-0000-0000-0000-000000000001"},
+    )
+    assert result["type"] is FlowResultType.FORM
+    assert result["step_id"] == "confirm"
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        user_input={},
+    )
+    assert result["title"] == "Husqvarna Automower"
+    assert result["handler"] == "husqvarna_automower_ble"
+
+    assert result["data"]["address"] == "00000000-0000-0000-0000-000000000001"
+
+
+async def test_no_devices(
+    hass: HomeAssistant,
+) -> None:
+    """Test missing device."""
+
+    inject_bluetooth_service_info(hass, AUTOMOWER_MISSING_SERVICE_SERVICE_INFO)
+    inject_bluetooth_service_info(hass, AUTOMOWER_UNSUPPORTED_GROUP_SERVICE_INFO)
+    await hass.async_block_till_done(wait_background_tasks=True)
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}
+    )
+    assert result["type"] is FlowResultType.FORM
+    assert result["step_id"] == "user"
+
+
+async def test_bluetooth(
+    hass: HomeAssistant,
+) -> None:
+    """Test bluetooth device discovery."""
+
+    inject_bluetooth_service_info(hass, AUTOMOWER_SERVICE_INFO)
+    await hass.async_block_till_done(wait_background_tasks=True)
+
+    result = next(iter(hass.config_entries.flow.async_progress_by_handler(DOMAIN)))
+    assert result["step_id"] == "confirm"
+    assert result["context"]["unique_id"] == "00000000-0000-0000-0000-000000000003"
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        user_input={},
+    )
+    assert result["title"] == "Husqvarna Automower"
+    assert result["handler"] == "husqvarna_automower_ble"
+
+    assert result["data"]["address"] == "00000000-0000-0000-0000-000000000003"
+    assert result["context"]["unique_id"] == "00000000-0000-0000-0000-000000000003"
```

Comment:
```suggestion
    assert result["result"]["unique_id"] == "00000000-0000-0000-0000-000000000003"
```

Reply:
```python
        result = hass.config_entries.flow.async_progress_by_handler(DOMAIN)[0]
        assert result["step_id"] == "confirm"
>       assert result["result"]["unique_id"] == "00000000-0000-0000-0000-000000000003"
E       KeyError: 'result'

tests/components/husqvarna_automower_ble/test_config_flow.py:84: KeyError
```

Reply:
That is strange, I think we need to make it so that it finalizes (aka, flowresulttype.CREATE_ENTRY